web/server/fastify/server.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { readFileSync } from "fs";
+import { fastify, FastifyHttpsOptions } from "fastify";
+import { fastifyConnectPlugin } from "@bufbuild/connect-fastify";
+import { cors as connectCors } from "@bufbuild/connect";
+import fastifyCors from "@fastify/cors";
+import routes from "../routes.js";
+import https from "https";
+import path from "path";
+import { fileURLToPath } from "url";
+import { dirname } from "path";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const HOST = "0.0.0.0";
+
+const CORS_OPTIONS = {
+  // Reflects the request origin. This should only be used for development.
+  // Production should explicitly specify an origin
+  origin: true,
+  methods: [...connectCors.allowedMethods],
+  allowedHeaders: [...connectCors.allowedHeaders],
+  exposedHeaders: [...connectCors.exposedHeaders],
+};
```

Comment:
You have to include the application-specific headers here. See https://github.com/bufbuild/connect-es/blob/b6187e9b2007bf178c0a6dea79cab3f30ff09c1a/packages/connect-node-test/src/helpers/testserver.ts#L177 for reference.